### DISCUSSION TITLE
Using event inputs in jobs.id.steps.uses

### DISCUSSION THREAD

**Discussion by unfor19** [community/community#26334] [Actions]
<p>According to the <a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstepsuses" rel="noopener nofollow ugc">workflow syntax</a> it’s possible to do this:</p>
<pre><code class="lang-yaml">on:
  workflow_dispatch:
    inputs:
      src_ref:
        description: Which branch to test?
        required: true
        default: master

jobs:
  my_first_job:
    steps:
      - name: My step
        uses: unfor19/install-aws-cli-action@v1 # &lt;-- static
</code></pre>
<p>But if I try to use a dynamic GitHub ref in <code>uses</code>, I get an error</p>
<pre><code class="lang-yaml">jobs:
  my_first_job:
    steps:
      - name: My step
        uses: unfor19/install-aws-cli-action@${{ github.event.inputs.src_ref }}
                                            # ^^^ Produces an error
</code></pre>
<p>Error:</p>
<blockquote>
<p>The workflow is not valid.  .github/workflows/test-action.yml (Line: 19, Col: 13): Unrecognized named-value: ‘github’. Located at position 1 within expression: github.event.inputs.src_ref</p>
</blockquote>
<p>Is there any way to pass a dynamic ref to <code>uses</code>?</p>

**Comment by Simran-B** [comment, marked as answer]
<p>No, it cannot be dynamic:</p>
<aside class="quote" data-post="2" data-topic="17466">
  <div class="title">
    <div class="quote-controls"></div>
    
    <a href="https://github.community/t/env-variables-in-uses/17466/2">Env variables in uses</a> <a class="badge-wrapper  bullet" href="/c/code-to-cloud/github-actions/41"><span class="badge-category-parent-bg" style="background-color: #6a737d;"></span><span class="badge-category-bg" style="background-color: #2188ff;"></span><span style="" data-drop-close="true" class="badge-category clear-badge" title="Discussion and questions about GitHub Actions—automation for all of your development workflows, now with world-class CI/CD.">GitHub Actions</span></a>
  </div>
  <blockquote>
    In order to enact policies like only using actions defined in the org or repo we can’t allow actions to dynamically change at runtime.  So using any sort of dynamic value in uses is not something we will be able to support.
  </blockquote>
</aside>

**Comment by unfor19** [comment]
<p>Thanks you <a class="mention" href="/simran-b">@Simran-B</a></p>
